src/replication.c

```diff
@@ -3216,14 +3222,44 @@ void setupMainConnForPsync(connection *conn) {
                                      "accept connections in read-write mode.\n");
         }
         dualChannelSyncHandlePsync();
-        return;
+        return C_OK;
     }
 
-error:
+    return psync_result;
+}
+
+/* Replication: Replica side.
+ * This connection handler fires after rdb-connection was initialized. We use it
+ * to adjust the replica main for loading incremental changes into the local buffer. */
+void setupMainConnForPsync(connection *conn) {
```

Comment:
What do you think about prefixing this function with `dualChan` as well?
```suggestion
void dualChanSetupMainConnForPsync(connection *conn) {
```

Reply:
Right, I just don't like the abbreviation Channel => Chan, I don't think it is very clean.